app/models/piece.rb

```diff
@@ -33,9 +33,21 @@ def is_obstructed?(x, y, x_target, y_target)
         end
       end
     elsif vertical_move?(x, y, x_target, y_target)
-      
+      (y...y_target).each do |y|
+        if game.square_occupied?(y)
```

Comment:
Since we redefined `square_occupied?` to take two parameters, this will need to be `game.square_occupied?(x, y)`. Line 29 will also need to be changed to include `(x, y)`.